libraries/chain/include/eosio/chain/transaction_context.hpp

```diff
@@ -74,6 +74,7 @@ namespace eosio { namespace chain {
          vector<action_receipt>        executed;
          flat_set<account_name>        bill_to_accounts;
          flat_set<account_name>        validate_ram_usage;
+         flat_map<account_name, int64_t> account_ram_delta; // reset for each action
```

Comment:
Instead of explicitly resetting for each action, move this field to `apply_context`.